secant/Features/BackupFlow/Views/RecoveryPhraseBackupValidationView.swift

```diff
@@ -0,0 +1,289 @@
+                    )
+                    .navigationLinkEmpty(
+                        isActive: viewStore.bindingForRoute(.failure),
+                        destination: { ValidationFailedView(store: store) }
+                    )
+                }
+                .frame(alignment: .top)
+            }
+            .applyScreenBackground()
+            .scrollableWhenScaledUp()
+            .navigationBarTitleDisplayMode(.inline)
+            .navigationTitle(Text("Verify Your Backup"))
+        }
+    }
+
+    @ViewBuilder func header(for viewStore: RecoveryPhraseValidationViewStore) -> some View {
+        VStack {
+            switch viewStore.step {
+            case .initial, .incomplete:
+                Text("Drag the words below to match your backed-up copy.")
+                    .bodyText()
+
+            case .complete:
+                completeHeader(for: viewStore.state)
+            }
+
+            viewStore.state.missingWordGrid()
+        }
+        .padding(.horizontal, 30)
+    }
+    
+    @ViewBuilder func completeHeader(for state: RecoveryPhraseValidationState) -> some View {
+        if state.isValid {
+            Text("Congratulations! You validated your secret recovery phrase.")
+                .bodyText()
+        } else {
+            Text("Your placed words did not match your secret recovery phrase")
+                .bodyText()
+        }
+    }
+}
+
+private extension RecoveryPhraseValidationState {
+    @ViewBuilder func missingWordGrid() -> some View {
+        let columns = Array(
+            repeating: GridItem(.flexible(minimum: 40, maximum: 120), spacing: 20),
+            count: 2
+        )
+
+        LazyVGrid(columns: columns, alignment: .center, spacing: 20) {
+            ForEach(0..<missingWordChips.count) { chipIndex in
+                PhraseChip(kind: missingWordChips[chipIndex])
+                    .makeDraggable()
+                    .frame(
+                        minWidth: 0,
+                        maxWidth: .infinity,
+                        minHeight: 30
+                    )
+            }
+        }
+        .padding(0)
+    }
+}
+
+extension RecoveryPhraseValidationState {
+    func wordsChips(
+        for groupIndex: Int,
+        groupSize: Int,
+        from wordGroup: RecoveryPhrase.Group
+    ) -> [PhraseChip.Kind] {
+        let validationWord = validationWords.first(where: { $0.groupIndex == groupIndex })
+
+        var chips: [PhraseChip.Kind] = []
+
+        for (index, word) in wordGroup.words.enumerated() {
+            if index == missingIndices[groupIndex] {
+                if let completedWord = validationWord?.word {
+                    chips.append(.unassigned(word: completedWord))
+                } else {
+                    chips.append(.empty)
+                }
+            } else {
+                chips.append(.ordered(position: (groupSize * groupIndex) + index + 1, word: word))
+            }
+        }
+        return chips
```

Comment:
You can get rid of the intermediate variable here if you just map over the words instead of using a for in loop.


```suggestion
        let validationWord = validationWords.first(where: { $0.groupIndex == groupIndex })
        return wordGroup.words.enumerated().map { index, word in
            if index == missingIndices[groupIndex] {
                if let completedWord = validationWord?.word {
                    return .unassigned(word: completedWord)
                } else {
                    return .empty
                }
            } else {
                return .ordered(position: (groupSize * groupIndex) + index + 1, word: word)
            }
        }
```

Reply:
This change is still valid to make, it's just getting rid of unnecessary code.